deeppavlov/core/models/torch_model.py

```diff
@@ -157,18 +161,31 @@ def load(self, fname: Optional[str] = None, *args, **kwargs) -> None:
                 log.info(f"Initializing `{self.__class__.__name__}` from saved.")
 
                 # firstly, initialize with random weights and previously saved parameters
-                self.init_from_opt(model_func)
+                if model_func:
+                    self.init_from_opt(model_func)
 
                 # now load the weights, optimizer from saved
                 log.info(f"Loading weights from {weights_path}.")
                 checkpoint = torch.load(weights_path, map_location=self.device)
-                self.model.load_state_dict(checkpoint["model_state_dict"], strict=False)
-                self.optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
+                model_state = checkpoint["model_state_dict"]
+                optimizer_state = checkpoint["optimizer_state_dict"]
+
+                # load a multi-gpu model on a single device
+                if not self.is_data_parallel and any(["module." in key for key in list(model_state.keys())]):
+                    model_state = {key.replace("module.", ""): val for key, val in model_state.items()}
+
+                strict_load_flag = any([key.endswith("embeddings.position_ids")
+                                        for key in checkpoint['model_state_dict']])
```

Comment:
need some comment for the future why this `strict_load_flag` is needed. Will this fix work with not-Bert models like Roberta or other encoder-only transformers?
TorchModel is a general class. It should not be dependent on the underlying model.